src/Host/Client/Impl/Extensions/RStringExtensions.cs

```diff
@@ -69,9 +68,61 @@ public static class RStringExtensions {
                         case 'v':
                             sb.Append("\v");
                             break;
-                        default:
+                        case 'x':
+                            // 1 to 2 hex digits, lowercase or uppercase
+                            if (i < s.Length - 1) {
+                                int val;
+                                if (HexCharToDecimal(s[i + 1], out val)) {
+                                    i++;
+                                    if (i < s.Length - 1) {
+                                        int nextVal;
+                                        if (HexCharToDecimal(s[i + 1], out nextVal)) {
+                                            i++;
+                                            val = (val << 4) | nextVal;
+                                        }
+                                    }
+
+                                    sb.Append(Char.ConvertFromUtf32(val));
+                                } else {
+                                    throw new FormatException("Expected hex character");
+                                }
+                            } else {
+                                throw new FormatException("Unexpected end of string");
+                            }
+                            break;
+                        case '\\':
+                            sb.Append(c);
+                            break;
+                        case '"':
                             sb.Append(c);
                             break;
+                        case '\'':
+                            sb.Append(c);
+                            break;
+                        default:
+                            if (c >= '0' && c <= '7') {
```

Comment:
Should be `<= '8'` per R sources. The rest of it doesn't change (i.e. `\8` is the same as `\10`).